[BUG] MacOS pytest doesn't work out of the box

**Describe the bug**
`pytest` doesn't work on `pip install -e ".[dev]" in clean conda env.
I get
```bash
...
exec(co, module.__dict__)
tests/tests/conftest.py:8: in <module>
    from cellfinder_core.download import models
src/cellfinder_core/__init__.py:13: in <module>
    raise PackageNotFoundError(
E   importlib.metadata.PackageNotFoundError: No package metadata was found for cellfinder tools cannot be invoked without tensorflow. Please install tensorflow into your environment to use cellfinder tools. For more information, please see https://github.com/brainglobe/brainglobe-meta#readme.
```

My guess is this is because our guard checks for the existence of a `tensorflow` package, but TensorFlow is called `tensorflow-macos` on Macs. 

I could be missing something though. I'm surprised we haven't noticed this elsewhere 🤔 

**To Reproduce**
See above: `pip install -e ".[dev]"` in clean conda env on Apple Silicon.

**Expected behaviour**
`pytest` should run the tests out-of-the box in a dev environment 

**Log file**
n/a

**Screenshots**
n/a

**Computer used (please complete the following information):**

- MacOS Apple M2

--- a/src/cellfinder_core/__init__.py
+++ b/src/cellfinder_core/__init__.py
@@ -10,12 +10,15 @@ except PackageNotFoundError as e:
 try:
     TF_VERSION = version("tensorflow")
 except PackageNotFoundError as e:
-    raise PackageNotFoundError(
-        f"cellfinder tools cannot be invoked without tensorflow. "
-        f"Please install tensorflow into your environment to use cellfinder tools. "
-        f"For more information, please see "
-        f"https://github.com/brainglobe/brainglobe-meta#readme."
-    ) from e
+    try:
+        TF_VERSION = version("tensorflow-macos")
+    except PackageNotFoundError as e:
+        raise PackageNotFoundError(
+            f"cellfinder tools cannot be invoked without tensorflow. "
+            f"Please install tensorflow into your environment to use cellfinder tools. "
+            f"For more information, please see "
+            f"https://github.com/brainglobe/brainglobe-meta#readme."
+        ) from e
 
 __author__ = "Adam Tyson, Christian Niedworok, Charly Rousseau"
 __license__ = "BSD-3-Clause"

